src/Stack/Package.hs

```diff
@@ -602,12 +602,36 @@ getBuildComponentDir Nothing = Nothing
 getBuildComponentDir (Just name) = parseRelDir (name FilePath.</> (name ++ "-tmp"))
 
 -- | Get all dependencies of the package (buildable targets only).
-packageDependencies :: PackageDescription -> Map PackageName VersionRange
-packageDependencies pkg =
+--
+-- Note that for Cabal versions 1.22 and earlier, there is a bug where
+-- Cabal requires dependencies for non-buildable components to be
+-- present. We're going to use GHC version as a proxy for Cabal
```

Comment:
Hmm, I don't see the reasoning of using a GHC version as a proxy for the cabal version.  AFAIU you can use Cabal 2 with GHC 7.8

Reply:
I considered going that route, but it's a much bigger diff, and gets into a bit of a circular issue, since we need to know the version of the Cabal library before we've finished calculating the build plan. The way I've written it, we'll occasionally (but rarely) get some unnecessary extra packages built if someone is using old GHC + new Cabal, but won't get any breakage (since you can't use old Cabal with a new GHC).

Reply:
I see!  Alright fair 'nough